File: lib/native/features/download/domain/entity/file_entity.dart
class FileEntity {
  String filename;
  String fileType;
  String thumbnailPath;
  String imagePath;
  String temporaryImagePath;

  FileEntity({
    required this.filename,
    required this.fileType,
    required this.thumbnailPath,
    required this.imagePath,
    required this.temporaryImagePath,
  });

  factory FileEntity.fromJson(Map<String, dynamic> json) => FileEntity(
      filename: json['filename'],
      fileType: json['fileType'],
      thumbnailPath: json['thumbnailPath'],
      imagePath: json['imagePath'],
      temporaryImagePath: json['temporaryImagePath']);

  Map<String, dynamic> toJson() => {
        'filename': filename,
        'fileType': fileType,
        'thumbnailPath': thumbnailPath,
        'imagePath': imagePath,
        'temporaryImagePath': temporaryImagePath,
      };
}

File: plugin/download/android/src/main/kotlin/com/example/perfect_corp_homework/download/model/FileEntity.kt
package com.example.perfect_corp_homework.download.model

import java.io.File

data class FileEntity(
    val filename: String,
    val fileType: String,
    val thumbnailPath: String,
    val imagePath: String,
    val temporaryImagePath: String
) {

    fun deleteTempFile() {
        File(temporaryImagePath).delete()
    }
}